examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,527 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+A simple tutorial that shows some features of the Temporal Memory.
+
+The following program has the purpose of presenting some
+basic properties of the Temporal Memory, in particular when it comes
+to how it handles high-order sequences.
+"""
+
+import numpy as np
+import random
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+from nupic.bindings.algorithms import TemporalMemory as TM
+
+def accuracy(curr, pred):
+  acc = 0
+  if np.count_nonzero(pred) > 0:
+    acc = float(np.dot(curr, pred))/float(np.count_nonzero(pred))
+  return acc
+
+def corruptVector(v1, noiseLevel, numActiveCols):
+  size = len(v1)
+  v2 = np.zeros(size, dtype="uint32")
+  bitsToSwap = int(noiseLevel * numActiveCols)
+  # Copy the contents of v1 into v2
+  for i in range(size):
+    v2[i] = v1[i]
+  for _ in range(bitsToSwap):
+    i = random.randrange(size)
+    if v2[i] == 1:
+      v2[i] = 0
+    else:
+      v2[i] = 1
+  return v2
+
+uintType = "uint32"
+random.seed(1)
+
+# After you have understood the concepts described in this script
+# you should try changing some of the parameters of the TM and
+# observe its behaviour.
+# For instance, try changing the number of cells per column,
```

Comment:
"For instance, try changing maxNewSynapseCount to 15 and see how it handles the noise. Or you can see what happens when you lower the thresholds significantly (e.g. to 3). You can also set the permanenceDecrement to zero and tell whether the TM is able to forget previous inputs."